modules/spark/docs/README.md

```diff
@@ -0,0 +1,81 @@
+# Spark module for PureConfig
+
+Adds support for selected [Spark](http://spark.apache.org/) classes to PureConfig.
+
+## Add pureconfig-spark to your project
+
+In addition to [core PureConfig](https://github.com/pureconfig/pureconfig), you'll need:
+
+```scala
+libraryDependencies += "com.github.pureconfig" %% "pureconfig-spark" % "@VERSION@"
+```
+
+Also, `pureconfig-spark` depends on `spark-sql` with `provided` scope.
+Spark libraries are generally added on runtime.
+This module has been tested on Spark 3 but it should also work for Spark 2.4 since basic datatype APIs should stay the same.
+Please note that we are only supporting Scala 2.12 for all Spark versions.
+
+To use the Spark module you need to import:
+```scala
+import pureconfig.module.spark._
+```
+
+## Supported classes
+
+* `org.apache.spark.sql.types.DataType`
+* `org.apache.spark.sql.types.StructType`
+
+## Example
+
+### Custom HOCON schema to/from Spark schema
+Setup custom schema case classes and converters between custom schema and Spark schema.
+```scala mdoc:silent
+import org.apache.spark.sql.types._
+import pureconfig._
+import pureconfig.generic.auto._
+import pureconfig.module.spark._
+
+case class MySchema(name: String, fields: List[Field])
+case class Field(name: String, dataType: DataType)
+
+def mySchemaToSparkSchema(schema: MySchema): StructType =
+  StructType(schema.fields.map { case Field(n, dt) => StructField(n, dt) })
```

Comment:
I'm curious about this example; why don't we also add a converter for `StructField` for completion? We can read from config objects with `name` and `dataType` fields, with the other two `nullable` and `metadata` as optional (choosing defaults for each).

Reply:
@ruippeixotog `StructField` can be derivable but it requires `Metadata`. Maybe I could try providing a converter for `Metadata` instead of fixing it to default?
The custom `Field` in the example is kind of trivial but it could also contain other settings meant for other data stores that may not be relevant for Spark.

Reply:
`Metadata` can be serialised as Json strings. I will also simplify my example to use `StructField` directly instead.

Reply:
@ruippeixotog resolving with f125110d9a0911fe774ef12425086e13263f4caa

Reply:
@ruippeixotog moved 5957f793b7f9fa116277c9c29a256bf12e4350ba